src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -2790,7 +2790,7 @@ class CommentOnStatementSegment(BaseSegment):
     https://www.postgresql.org/docs/13/sql-comment.html
     """
 
-    type = "comment_on_statement"
+    type = "comment_clause"
```

Comment:
Modify segment type to match the type name used in other dialects (`ansi`, `exasol`, `teradata`)